src/error.ts

```diff
@@ -403,6 +414,32 @@ export class MongoGridFSChunkError extends MongoRuntimeError {
   }
 }
 
+/**
+ * An error generated when a **parsable** unexpected response comes from the server.
+ * This is generally an error where the driver in a state expecting a certain behavior to occur in
+ * the next message from MongoDB but it receives something else.
+ * This error **does not** represent an issue with wire message formatting.
+ *
+ * #### Example
+ * When an operation fails, it is the driver's job to retry it. It must perform serverSelection
+ * again to make sure that it attempts the operation against a server in a good state. If server
+ * selection returns a server that does not support retryable operations, this error is used.
+ * This scenario is unlikely as retryable support would also have been determined on the first attempt
+ * but it is possible the state change could report a selectable server that does not support retries.
+ *
+ * @public
+ * @category Error
+ */
+export class MongoUnexpectedServerResponseError extends MongoRuntimeError {
```

Comment:
I haven't looked closely,  but I would guess that even before this PR our generated "error tree" svg used in our error documentation is out of date (see https://github.com/mongodb/node-mongodb-native/tree/main/docs/charts).  

It's out of scope to in your work to get them back into sync but would you mind adding your new error to the generate script, generating a new SVG and including those as changes in your PR?